build: fix adev first party linking and re-enable tests

devversion opened the pull request:
Follow-up to https://github.com/angular/angular/pull/55282

This commit attempts to finally fix the long-standing first-party package linking issue with the rather tricky `rules_nodejs` toolchain.

I've verified that no version of e.g. `@angular/core` ends up in the Bazel sandbox. This is achieved by also filtering transitive Angular deps for first-party linked packages. e.g. `@angular/docs`.

In addition, `@angular/docs` accidentally ended up bundling parts of Angular core because it relied on an entry-point that was not part of the "well known externals". As part of the ongoing `ng_package` update/rewrite, we should look into disabling bundling of ANY external dependency/module. This is possible because we use relative imports inside APF packages as of recently!

This commit should allow us to develop and continue new compiler features, without having to temporarily (or longer) disable all `angular.dev` unit tests!

devversion committed:
build: fix adev first party linking and re-enable tests

This commit attempts to finally fix the long-standing first-party
package linking issue with the rather tricky `rules_nodejs` toolchain.

I've verified that no version of e.g. `@angular/core` ends up in the
Bazel sandbox. This is achieved by also filtering transitive Angular
deps for first-party linked packages. e.g. `@angular/docs`.

In addition, `@angular/docs` accidentally ended up bundling parts of
Angular core because it relied on an entry-point that was not part of
the "well known externals". As part of the ongoing `ng_package`
update/rewrite, we should look into disabling bundling of ANY external
dependency/module. This is possible because we use relative imports
inside APF packages as of recently!

This commit should allow us to develop and continue new compiler
features, without having to temporarily (or longer) disable all
`angular.dev` unit tests!

Fixes #54858.

alan-agius4 commented on .github/workflows/ci.yml:
```diff
@@ -107,10 +107,8 @@ jobs:
           google_credential: ${{ secrets.RBE_TRUSTED_BUILDS_USER }}
```
OOC, is https://github.com/angular/angular/blob/36ae560765ab5a7e1b121b0ca3e6c1e8e77cc34b/package.json#L165 still needed?

devversion replied:
Yeah, seems unused. I don't know why it even is there

josephperrott reviewed (approved):
LGTM

alan-agius4 reviewed (approved):
LGTM

devversion committed:
build: remove unnecessary feature detection that breaks with shared chunks

Since APF can contain shared chunks, where e.g. `eetemplate` lives, the
`coreHasSymbol` check is no longer reliable. Right now it even prevents
the version range check (that is reliably working) from running because
we detect a `index.d.ts` file but simply don't find the requested
symbol in there (we don't expand exports via type checker).

devversion commented on packages/compiler-cli/src/ngtsc/core/src/compiler.ts:
```diff
@@ -1035,14 +1034,12 @@ export class NgCompiler {
+    // writable signals for type-checking. Only Angular versions greater than 17.2 have the necessary
+    // symbols to type check signals in two-way bindings. We also allow version 0.0.0 in case somebody is
     // using Angular at head.
     let allowSignalsInTwoWayBindings =
```
Alan logged into the GitHub VM and made a test fail, so I had to re-push. Done!

alan-agius4 reviewed (approved):
LGTM

devversion commented:
This PR was merged into the repository by commit abe96914f589ef236dffbd4f6f6bf27ff4f7e80b.

The changes were merged into the following branches: main
